src/cpp/asyncencode.cpp

```diff
@@ -28,17 +28,22 @@
 class AsyncEncodeWorker : public Nan::AsyncWorker {
     public:
         AsyncEncodeWorker(Nan::Callback *callback, int instance_descriptor_id,
-                int k, int m, char *orig_data, int orig_data_size) :
+                int k, int m, v8::Local<v8::Object> &orig_data,
+                int orig_data_size, int persistent_flag) :
             Nan::AsyncWorker(callback),
             _instance_descriptor_id(instance_descriptor_id),
             _k(k),
             _m(m),
-            _orig_data(orig_data),
-            _orig_data_size(orig_data_size) {}
+            _orig_data_size(orig_data_size),
+            _persistent_flag(persistent_flag) {
+                if (_persistent_flag == 1) {
+                    // make sure orig_data isn't GC'ed
+                    SaveToPersistent("orig_data", orig_data);
+                }
+                _orig_data = node::Buffer::Data(orig_data);
+            }
 
-        ~AsyncEncodeWorker() {
-            delete _orig_data;
-        }
+        ~AsyncEncodeWorker() {}
```

Comment:
I don't really understand your question. Is it about `_persistent_flag`? If yes, we don't need to do that